src/providers/fault-count/fault-count.ts

```diff
@@ -122,6 +135,96 @@ export class FaultCountProvider {
     };
   }
 
+  private getVehicleChecksFaultCountCatCE = (vehicleChecks: CatCEUniqueTypes.VehicleChecks): VehicleChecksScore => {
```

Comment:
These functions could probably be abstracted for readability. There's a bit of reading in them but I'd be happy to approve as the logics sound.

Reply:
Agreed there is some refactoring work needed on this provider to deal with the duplication/massive file size. I'll take a look at what could be done :)